pkg/kube/controllers/extendedjob/job_creator.go

```diff
@@ -106,6 +104,45 @@ func (j jobCreatorImpl) Create(ctx context.Context, eJob ejv1.ExtendedJob, names
 		}
 	}
 
+	// Fetch all secrets
+	tokenSecret := corev1.Secret{}
+	secretList := &corev1.SecretList{}
+	err = j.client.List(ctx, secretList, client.InNamespace(namespace))
+	if err != nil {
+		return false, errors.Wrapf(err, "could not get secret list related to ejob %s.", eJob.Name)
+	}
+	for _, secret := range secretList.Items {
+		annotations := secret.GetAnnotations()
+		annotation, ok := annotations["kubernetes.io/service-account.name"]
+		if ok {
+			if annotation == serviceAccountName {
+				tokenSecret = secret
+				break
+			}
+		}
+	}
```

Comment:
What about the case where no token secret was found? Or is it guaranteed that it will always be there?

Reply:
It's guaranteed by line 93 when the service account is created.